File: front/client/api/models/translation.ts
export interface Translation {
	messageID:      number;
	lang:           string;
	title:          string;
	text:           string;
	createdAt:      string;
	updatedAt:      string;
}

const EmptyTranslation: Translation = Object.freeze({
	messageID:  0,
	lang:       "",
	createdAt:  "",
	updatedAt:  "",
	title:      "",
	text:       "",
})

export default function mapTranslationFromProto(translation?: any): Translation {
	if (!translation) {
		return EmptyTranslation
	}

	let createdAt = "0"
	if (translation.created_at) {
		createdAt = new Date(Math.floor(translation.created_at / ns_in_ms)).toLocaleString()
	}

	let updatedAt = "0"
	if (translation.updated_at) {
		updatedAt = new Date(Math.floor(translation.updated_at / ns_in_ms)).toLocaleString()
	}

	const res = {
		createdAt:   createdAt,
		updatedAt:   updatedAt,
		text:        translation.text,
		title:       translation.title,
		messageID:   translation.message,
		lang:        translation.lang,
	}

	return res
}

export { EmptyTranslation }

File: server/messages/pkg/model/translation.go
package model

type (
	Translation struct {
		// TODO UserID
		MessageID           int64               `json:"message"`
		Lang                string              `json:"lang"`
		Title               string              `json:"title"`
		Text                string              `json:"text"`
		CreatedAt           string              `json:"created_at"`
		UpdatedAt           string              `json:"updated_at"`
	}

	TranslationPreview struct {
		MessageID           int64               `json:"message"`
		Lang                string              `json:"lang"`
		Title               string              `json:"title"`
		CreatedAt           string              `json:"created_at"`
		UpdatedAt           string              `json:"updated_at"`
	}

	SendTranslationRequest struct {
		MessageID           int64               `json:"message"`
		Lang                string              `json:"lang"`
		Title               string              `json:"title"`
		Text                string              `json:"text"`
	}

	SendTranslationResponse struct {
		Description         string              `json:"description"`
	}

	UpdateTranslationRequest struct {
		MessageID           int64               `json:"message"`
		Lang                string              `json:"lang"`
		Title               *string             `json:"title,omitempty"`
		Text                *string             `json:"text,omitempty"`
	}

	UpdateTranslationResponse struct {
		Description         string              `json:"description"`
	}

	DeleteTranslationRequest struct {
		MessageID           int64               `json:"message"`
		Lang                string              `json:"lang"`
	}

	DeleteTranslationResponse struct {
		Description         string              `json:"description"`
	}

	ReadTranslationRequest struct {
		MessageID           int64               `json:"message"`
		Lang                string              `json:"lang"`
		Name                *string             `json:"name,omitempty"`
	}

	ReadTranslationResponse struct {
		Translation         *Translation        `json:"translation"`
	}

	ListTranslationsRequest struct {
		MessageID           int64               `json:"message"`
		Name                string              `json:"name,omitempty"`
	}

	ListTranslationsResponse struct {
		Translations        []*Translation      `json:"translations"`
	}
)